packages/language-service/src/lib/service.ts

```diff
@@ -233,9 +236,68 @@ export class StylableLanguageService {
         return null;
     }
 
-    public onDocumentFormatting() {
-        // no op
-        return null;
+    public onDocumentFormatting(
+        filePath: string,
+        options: { tabSize: number; insertSpaces: boolean }
+    ): TextEdit[] {
+        const stylableFile = this.readStylableFile(filePath);
+
+        if (stylableFile && stylableFile.stat.isFile()) {
+            const doc = TextDocument.create(
+                URI.file(filePath).toString(),
+                'stylable',
+                stylableFile.stat.mtime.getTime(),
+                stylableFile.content
+            );
+
+            return this.getDocumentFormatting(
+                doc,
+                { start: 0, end: doc.getText().length },
+                normalizeVSCodeFormattingOptions(options)
+            );
+        }
+
+        return [];
+    }
+
+    public onDocumentRangeFormatting(
+        filePath: string,
+        offset: { start: number; end: number },
+        options: { tabSize: number; insertSpaces: boolean }
+    ) {
+        const stylableFile = this.readStylableFile(filePath);
```

Comment:
readStylableFile ignores untitled, which is not ideal for formatting. consider making this provider a bit more flexible, as it only needs the content.